src/fatx_fat.c

```diff
@@ -103,40 +209,36 @@ int fatx_get_fat_entry_type(struct fatx_fs *fs, fatx_fat_entry entry)
     if (fs->fat_type == FATX_FAT_TYPE_16)
     {
         entry &= FATX_FAT16_ENTRY_MASK;
-
-        if (entry == 0)
-            return FATX_CLUSTER_AVAILABLE;
-
-        if (entry == 1)
-            return FATX_CLUSTER_RESERVED;
-
-        if (entry >= FATX_FAT_RESERVED_ENTRIES_COUNT && entry <= 0xfff6)
-            return FATX_CLUSTER_DATA;
-
-        if (entry == 0xfff7)
-            return FATX_CLUSTER_BAD;
-
-        if (entry >= 0xfff8 && entry <= 0xffff)
-            return FATX_CLUSTER_END;
+        switch (entry)
+        {
+            case 0x0000:
+                return FATX_CLUSTER_AVAILABLE;
+            case 0xfff0:
+                return FATX_CLUSTER_RESERVED;
+            case 0xfff7:
+                return FATX_CLUSTER_BAD;
+            case 0xfff8:
+                return FATX_CLUSTER_DATA;
+            case 0xffff:
+                return FATX_CLUSTER_END;
+        }
     }
     else
     {
         entry &= FATX_FAT32_ENTRY_MASK;
-
-        if (entry == 0)
-            return FATX_CLUSTER_AVAILABLE;
-
-        if (entry == 1)
-            return FATX_CLUSTER_RESERVED;
-
-        if (entry >= FATX_FAT_RESERVED_ENTRIES_COUNT && entry <= 0x0ffffff6)
-            return FATX_CLUSTER_DATA;
-
-        if (entry == 0x0ffffff7)
-            return FATX_CLUSTER_BAD;
-
-        if (entry >= 0x0ffffff8 && entry <= 0x0fffffff)
-            return FATX_CLUSTER_END;
+        switch (entry)
+        {
+            case 0x00000000:
+                return FATX_CLUSTER_AVAILABLE;
+            case 0xfffffff0:
+                return FATX_CLUSTER_RESERVED;
+            case 0xfffffff7:
+                return FATX_CLUSTER_BAD;
+            case 0xfffffff8:
+                return FATX_CLUSTER_DATA;
+            case 0xffffffff:
+                return FATX_CLUSTER_END;
+        }
     }
```

Comment:
TBH this whole func seemed very out-of-spec? Maybe it was written based on non-xbox FAT documentation. I noticed when I was trying to write the root directory entry.

With my changes I tried to stick to the facts and simplify.